src/Token/AccessTokenInterface.php

```diff
@@ -69,4 +70,20 @@ public function __toString();
      * @return array
      */
     public function jsonSerialize();
+
+    /**
+     * Sets the clock.
+     *
+     * @param Clock $clock a clock.
+     *
+     * @return void
+     */
+    public function setClock(Clock $clock);
```

Comment:
Typically, we want interfaces to define things we can fetch, not necessarily how they are set. The way something is set could be left open to each implementation, but we'd want to ensure that all implementations have a way to get the clock (i.e., `getClock(): Clock`).

Reply:
sorry, I dont understand what you're saying here. If this is about clock interfaces, im all for it.

Reply:
I'm saying I would prefer to define a `getClock()` method on the `AccessTokenInterface`, rather than a `setClock()` method.

Reply:
The changes to this interface don't make sense. An access token does not have a "clock", it has an expiry date/time. I think the only reasonable modification here would be `AccessTokenInterface::getExpiresDate()` that returns `DateTimeImmutable`. Given that `getExpires` returns a UNIX timestamp already, it makes no sense for the access token to have a separate clock.